instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/HistoryRecordingStateChangeObserverTest.kt

```diff
@@ -0,0 +1,508 @@
+            )
+        )
+    }
+
+    private fun getOffRouteLocation(originLocation: Point): Location =
+        mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = originLocation.latitude() + 0.002
+            longitude = originLocation.longitude()
+        }
+
+    private fun createMapboxNavigation() {
+        mapboxNavigation = MapboxNavigationProvider.create(
+            NavigationOptions.Builder(activity)
+                .accessToken(getMapboxAccessTokenFromResources(activity))
+                .build()
+        )
+    }
+
+    private fun observeHistoryRecordingEvents(
+        eventsChannel: Channel<HistoryRecordingStateChangeEvent>
+    ) {
+        val observer = object : HistoryRecordingStateChangeObserver {
+
+            override fun onShouldStartRecording(state: NavigationSessionState) {
+                CoroutineScope(Dispatchers.Main.immediate).launch {
+                    eventsChannel.send(
+                        HistoryRecordingStateChangeEvent(
+                            HistoryRecordingStateChangeEventType.START,
+                            state::class
+                        )
+                    )
+                }
+
+            }
+
+            override fun onShouldStopRecording(state: NavigationSessionState) {
+                CoroutineScope(Dispatchers.Main.immediate).launch {
+                    eventsChannel.send(
+                        HistoryRecordingStateChangeEvent(
+                            HistoryRecordingStateChangeEventType.STOP,
+                            state::class
+                        )
+                    )
+                }
+            }
+
+            override fun onShouldCancelRecording(state: NavigationSessionState) {
+                CoroutineScope(Dispatchers.Main.immediate).launch {
+                    eventsChannel.send(
+                        HistoryRecordingStateChangeEvent(
+                            HistoryRecordingStateChangeEventType.CANCEL,
+                            state::class
+                        )
+                    )
+                }
+            }
+        }
+        mapboxNavigation.registerHistoryRecordingStateChangeObserver(observer)
+    }
+
+    private suspend fun awaitStopActiveGuidanceRecording(
+        routes: List<NavigationRoute>
+    ) = suspendCancellableCoroutine<String?> { continuation ->
+        val observer = object : HistoryRecordingStateChangeObserver {
+            override fun onShouldStartRecording(state: NavigationSessionState) {
+                mapboxNavigation.historyRecorder.startRecording()
+            }
+
+            override fun onShouldStopRecording(state: NavigationSessionState) {
+                mapboxNavigation.historyRecorder.stopRecording { filePath ->
+                    if (state is NavigationSessionState.ActiveGuidance) {
+                        mapboxNavigation.unregisterHistoryRecordingStateChangeObserver(this)
+                        continuation.resume(filePath) {}
```

Comment:
```suggestion
                        continuation.resume(filePath)
```
The same below. 
You should also add
```kotlin
import kotlin.coroutines.resume
```